store/storetest/thread_store.go

```diff
@@ -1226,3 +1227,173 @@ func testMarkAllAsReadByChannels(t *testing.T, ss store.Store) {
+			UserId:    u2.Id,
+		})
+		require.NoError(t, err)
+		post3, err := ss.Post().Save(&model.Post{
+			ChannelId: channel.Id,
+			UserId:    u3.Id,
+		})
+		require.NoError(t, err)
+		threadStoreCreateReply(t, ss, channel.Id, post1.Id, post1.UserId, 2000)
+		threadStoreCreateReply(t, ss, channel.Id, post1.Id, post1.UserId, 2000)
+
+		threadStoreCreateReply(t, ss, channel.Id, post2.Id, post2.UserId, 2000)
+		threadStoreCreateReply(t, ss, channel.Id, post2.Id, post2.UserId, 2000)
+		threadStoreCreateReply(t, ss, channel.Id, post3.Id, post3.UserId, 2000)
+		opts := store.ThreadMembershipOpts{
+			Following:             true,
+			IncrementMentions:     false,
+			UpdateFollowing:       true,
+			UpdateViewedTimestamp: false,
+			UpdateParticipants:    false,
+		}
+
+		// create threadmemberships entries.
+		_, err = ss.Thread().MaintainMembership(post1.UserId, post1.Id, opts)
+		require.NoError(t, err)
+		_, err = ss.Thread().MaintainMembership(post2.UserId, post2.Id, opts)
+		require.NoError(t, err)
+		_, err = ss.Thread().MaintainMembership(post2.UserId, post3.Id, opts)
+		require.NoError(t, err)
+
+		//  get top threads by user
+		topThreadsByUser1, err := ss.Thread().GetTopThreadsForUserSince(team.Id, post1.UserId, 12, 0, limit)
+		require.NoError(t, err)
+		topThreadsByUser2, err := ss.Thread().GetTopThreadsForUserSince(team.Id, post2.UserId, 12, 0, limit)
+		require.NoError(t, err)
+		// require length of top threads by users to be 1,2 respectively
+		require.Len(t, topThreadsByUser1.Items, 1)
+		require.Len(t, topThreadsByUser2.Items, 2)
+
+		// require first element of topThreadsByUser1 to be post1 with 2 replyCount=2
+		require.Equal(t, topThreadsByUser1.Items[0].PostId, post1.Id)
+		require.Equal(t, topThreadsByUser1.Items[0].ReplyCount, int64(2))
+		require.Equal(t, topThreadsByUser1.Items[0].Message, post1.Message)
+		require.Equal(t, topThreadsByUser1.Items[0].UserId, post1.UserId)
+		require.Equal(t, topThreadsByUser1.Items[0].UserInformation.Id, post1.UserId)
+		// require elements of topThreadsByUser2 to be post2 and post3 respectively
+		require.Equal(t, topThreadsByUser2.Items[0].PostId, post2.Id)
+		require.Equal(t, topThreadsByUser2.Items[0].ReplyCount, int64(2))
+		require.Equal(t, topThreadsByUser2.Items[0].Message, post2.Message)
+		require.Equal(t, topThreadsByUser2.Items[0].UserId, post2.UserId)
+		require.Equal(t, topThreadsByUser2.Items[0].UserInformation.Id, post2.UserId)
+
+		require.Equal(t, topThreadsByUser2.Items[1].PostId, post3.Id)
+		require.Equal(t, topThreadsByUser2.Items[1].ReplyCount, int64(1))
+		require.Equal(t, topThreadsByUser2.Items[1].Message, post3.Message)
+		require.Equal(t, topThreadsByUser2.Items[1].UserId, post3.UserId)
+		require.Equal(t, topThreadsByUser2.Items[1].UserInformation.Id, post3.UserId)
+
+		// require topThreads[i].Post is not null
+		require.Equal(t, topThreadsByUser1.Items[0].Post.Id, post1.Id)
+		require.Equal(t, topThreadsByUser2.Items[1].Post.Id, post3.Id)
+	})
+
+}
```

Comment:
There are some cases that we should include here (I may have missed them if they're there):
1. results don't include threads from a different team
2. results don't include private channels a user is not a member of
3. results don't include DM threads
4. results only include threads within the given time frame
5. something testing the pagination
If these cases (and any others you can think of) are covered in the API tests that's totally fine. We just want to ensure that all of the logical cases are covered.

Reply:
This PR only had test case 2 covered.

But [this commit](https://github.com/mattermost/mattermost-server/pull/20195/commits/07985a3dda589769fae3c09b84d895ca9cd32a60) adds the other test cases you mentioned.